File: leetcode/java/0108. Convert Sorted Array to Binary Search Tree.java
package leetcode.java;

// Definition for a binary tree node.
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

// Time complexity: O(N), where N is the number of nodes
// Space complexity: O(N)

class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return this.insert(nums, 0, nums.length-1);
    }

    public TreeNode insert(int[] nums, int left, int right) {
        if (right < left) {
            return null;
        }

        int mid = (left + right) / 2;
        TreeNode node = new TreeNode(nums[mid]);

        node.left = this.insert(nums, left, mid-1);
        node.right = this.insert(nums, mid+1, right);
        
        return node;
    }
}

File: leetcode/python/0108. Convert Sorted Array to Binary Search Tree.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

# Time complexity: O(N), where N is the number of nodes
# Space complexity: O(N)

class Solution(object):
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        if len(nums) > 0:
            mid = len(nums) # 2
            res = TreeNode(nums[mid])

            res.left = self.sortedArrayToBST(nums[0:mid])
            res.right = self.sortedArrayToBST(nums[mid+1:])

            return res